### ISSUE TITLE
Discovery assumes that we got a Solr connection, leading to possible NPE

### ISSUE THREAD

**Issue by mwoodiupui** [DSpace/DSpace#3172]
**Describe the bug**
`SolrSearchCore.getSolr()` assumes that `initSolr()` has placed a non-null reference in field `solr` and unconditionally dereferences it.  If `discovery.search.server` is a syntactically correct URL but points to a down Solr instance, or no Solr instance at all, this yields NPE, which is not very helpful.  Stack trace from @hardyoyo attached:  [npe.txt](https://github.com/DSpace/DSpace/files/6078958/npe.txt)

**To Reproduce**
1. Configure `discovery.search.server` to a valid but useless URL.
2. Trigger a Discovery search.  For example:  try to create a new Collection, given an existing Community.

**Expected behavior**
`getSolr()` should throw a more specific exception which informs the catcher that Solr could not be contacted.  It should be an exception that can be passed to the UI so that a useful error banner can be shown, facilitating the detection and correction of the problem with contacting Solr.

**Related work**
Link to any related tickets or PRs here.

**Comment by mwoodiupui** [comment]
Maybe this should be expanded to examine all four Solr connections (search, authority, OAI PMH, statistics) for similar issues.

**Comment by tdonohue** [comment]
@mwoodiupui : I agree with you on this.  If you plan to work on this, let me know as I can always assign it back to you.  Otherwise, I'm going to leave this in the backlog while I determine where best to place this (possibly 7.0, but also could be 7.1)

**Comment by tdonohue** [comment]
Moving this to the 7.1 board for now, as it seems low priority for now.  However, @mwoodiupui if you find time to fix this in 7.0, we can move it up.
